File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/service/mapping/ExpenseItemMapper.java
package com.github.arhor.simple.expense.tracker.service.mapping;

import org.mapstruct.InheritInverseConfiguration;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import com.github.arhor.simple.expense.tracker.config.mapping.MapStructConfig;
import com.github.arhor.simple.expense.tracker.data.model.ExpenseItem;
import com.github.arhor.simple.expense.tracker.model.ExpenseItemDTO;

@Mapper(config = MapStructConfig.class)
public interface ExpenseItemMapper {

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "expenseId", ignore = true)
    ExpenseItem mapToEntity(ExpenseItemDTO dto);

    @InheritInverseConfiguration
    ExpenseItemDTO mapToDTO(ExpenseItem entity);
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/mapping/ExpenseItemMapper.kt
package com.github.arhor.simple.expense.tracker.service.mapping

import com.github.arhor.simple.expense.tracker.data.model.ExpenseItem
import com.github.arhor.simple.expense.tracker.model.ExpenseItemRequestDTO
import com.github.arhor.simple.expense.tracker.model.ExpenseItemResponseDTO
import org.mapstruct.InheritInverseConfiguration
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper(config = MapstructCommonConfig::class)
interface ExpenseItemMapper {

    @Mapping(target = "id", ignore = true)
    fun mapToEntity(dto: ExpenseItemRequestDTO, expenseId: Long): ExpenseItem

    @InheritInverseConfiguration
    fun mapToDTO(entity: ExpenseItem): ExpenseItemResponseDTO
}
